src/Features/Core/Portable/Workspace/ProjectCacheService.cs

```diff
@@ -135,24 +129,6 @@ private bool PartOfP2PReferences(ProjectId key)
             return false;
         }
 
-        [return: NotNullIfNotNull("instance")]
-        public T? CacheObjectIfCachingEnabledForKey<T>(ProjectId key, ICachedObjectOwner owner, T? instance) where T : class
```

Comment:
no longer ever called, so entirely removed (along with any code only called/used from this).